src/index.ts

```diff
@@ -61,6 +72,129 @@ const plugin: JupyterFrontEndPlugin<void> = {
 
     const sharingService = new SharingService(apiUrl);
 
+    async function handleNotebookSave(
+      notebookPanel: NotebookPanel,
+      isManualShare: boolean = false
+    ) {
+      const notebookContent = notebookPanel.context.model.toJSON() as INotebookContent;
+
+      // Check if notebook has already been shared
+      const isAlreadyShared =
+        notebookContent.metadata &&
+        typeof notebookContent.metadata === 'object' &&
+        'sharedId' in notebookContent.metadata;
+
+      if (isAlreadyShared && !isManualShare) {
+        try {
+          const sharedId = notebookContent.metadata.sharedId as string;
+          console.log('Updating notebook:', sharedId);
+
+          await sharingService.token;
```

Comment:
If I'm reading this correctly, this is to ensure that the user is authenticated; if not, `sharingService.authenticate()` is called under the hood.

But this already happens automatically in `sharingService.update()`, so I think we can remove this?

Reply:
I think this was resolved by your changes in agriyakhetarpal/jupytereverywhere-jupyterlite-extension#1. Thanks!